kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/signature/ContainerSignatureValidationService.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.container.signature;
+
+import java.util.Optional;
+import org.eclipse.kura.configuration.Password;
+import org.eclipse.kura.container.orchestration.ImageInstanceDescriptor;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Interface representing a service for validating the signature of a container image
+ *
+ * @since 2.3
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+@ProviderType
+public interface ContainerSignatureValidationService {
+
+    /**
+     * Verifies the signature of a container image using the provided trust anchor. The trust anchor format depends on
+     * the signature format. For example, if the signature was generated with Cosing, the trust anchor is a ECDSA public
+     * key in PEM format. Other signature formats may require different trust anchors.
+     *
+     * If the signature is not included in a transparency log, the verification will fail unless the
+     * verifyInTransparencyLog is set to false.
+     *
+     * If the image is signed with a different protocol the verification will fail.
+     *
+     * If the device running the verification has no internet access, the verification will fail.
```

Comment:
Can we state this or it is implementation specific?

Reply:
This is our design choice: if we don't have internet access we cannot perform the verification and thus return `false`.

Reply:
Addressed in 4ff066f1ed4b4ab2b2764608ed33b3cf95f972c7 as per our discussion